lesson_7/vlad_butulan/work3.py

```diff
@@ -0,0 +1,25 @@
+# Create a decorator called register that will update a list called print_registry with all the decorated functions names
+print_registry = []
+
+def register(fct):
+    def inner(*args):
+        print_registry.append(fct.__name__)
+        print(fct(*args))
+    return inner
```

Comment:
This decorator does add the function to the registry but also does some additional stuff; I challenge you to find the bug and refactor the decorator and make it simpler.

Hint: check the returned values for the decorated functions

Reply:
I see that the returned values are only the function names
Can't find the additional stuff

Reply:
There is no need for the inner function here; you could have it as follows:
```python 
def register(fct):
    print_registry.append(fct.__code__.co_name)
    return fct(*args)
```